pkg/services/pluginsintegration/pipeline/pipeline.go

```diff
@@ -29,7 +29,7 @@ func ProvideDiscoveryStage(cfg *config.Cfg, pf finder.Finder, pr registry.Servic
 				plugins.TypeDataSource, plugins.TypeApp, plugins.TypePanel, plugins.TypeSecretsManager,
 			}),
 			func(ctx context.Context, _ plugins.Class, b []*plugins.FoundBundle) ([]*plugins.FoundBundle, error) {
-				return discovery.NewDuplicatePluginFilterStep(pr).Filter(ctx, b)
+				return NewDuplicatePluginIDFilterStep(pr).Filter(ctx, b)
```

Comment:
Correct me if I am wrong but in the same pipeline, if there are two duplicated plugins, it won't find it as duplicated since they have not yet being registered, right?. Not that it's a big deal but wanted to know if I am understanding this properly.

Reply:
Well since the registry is shared amongst all pipelines, this will ensure no duplicates will be found across all plugin loading.

Reply:
yes, but in the same pipeline execution (e.g. when loading core plugins) that bundle has not been registered yet so for example, 2 identical core plugins would not be detected as duplicated, no?

Reply:
Ah sorry yes for that case you are right. That has always been the case, but no reason why we shouldn't patch this up at some point. We could probably add logic to the same step (as long as we know we're loading as a bundle and not as separate sources)